### ISSUE TITLE
xgboostjob_v1alpha1_iris_train_local example failed with XGBoostJob v1 

### ISSUE THREAD

**Issue by runzhliu** [kubeflow/xgboost-operator#92]
[The v1alpha1 example](https://github.com/kubeflow/xgboost-operator/blob/master/config/samples/xgboost-dist/xgboostjob_v1alpha1_iris_train_local.yaml) failed although I change the `apiVersion` from `xgboostjob.kubeflow.org/v1alpha1` to `xgboostjob.kubeflow.org/v1`.
Is there any available v1 example? :)

```
time="2020-06-12T07:21:33Z" level=info msg="Need to create new pod: worker-0" job=default.xgboost-dist-iris-test-train-local replica-type=worker uid=3b2b036e-ac51-11ea-8e12-48d539a94a19
E0612 07:21:33.754640       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 433 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1533f60, 0x26d96c0)
	/go/src/github.com/kubeflow/xgboost-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/kubeflow/xgboost-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1533f60, 0x26d96c0)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/kubeflow/common/pkg/controller.v1/control.RealPodControl.createPods(0x0, 0x0, 0x192c660, 0xc00067d240, 0x0, 0x0, 0xc00051cc99, 0x7, 0xc00043f800, 0x1906460, ...)
	/go/src/github.com/kubeflow/xgboost-operator/vendor/github.com/kubeflow/common/pkg/controller.v1/control/pod_control.go:149 +0x21b
github.com/kubeflow/common/pkg/controller.v1/control.RealPodControl.CreatePodsWithControllerRef(0x0, 0x0, 0x192c660, 0xc00067d240, 0xc00051cc99, 0x7, 0xc00043f800, 0x1906460, 0xc0009d6a80, 0xc00e3a82d0, ...)
	/go/src/github.com/kubeflow/xgboost-operator/vendor/github.com/kubeflow/common/pkg/controller.v1/control/pod_control.go:102 +0xe9
github.com/kubeflow/common/pkg/controller.v1/common.(*JobController).createNewPod(0xc000ab4000, 0x16b42c0, 0xc0009d6a80, 0xc00df0a2d8, 0x6, 0x170ca33, 0x1, 0xc00043e300, 0x0, 0xc00dd74120, ...)
```

**Comment by merlintang** [comment]
we need to update the example to v1

**Comment by Jeffwan** [comment]
crds have been updated to v1. #96 
I will update examples in a separate PR. I need to run some tests

**Comment by Focke-Wulf** [comment]
> crds have been updated to v1. #96
> I will update examples in a separate PR. I need to run some tests

any updated？ I also got this problem.

**Comment by Focke-Wulf** [comment]
I also got this problem after I update CRD to v1
<img width="1105" alt="2021-01-28 1 33 16" src="https://user-images.githubusercontent.com/26343300/106567513-e9e67500-656c-11eb-8cc0-1a519236d1a2.png">

here is the error logs

**Comment by hemantha-kumara** [comment]
any updates?

> crds have been updated to v1. #96
> I will update examples in a separate PR. I need to run some tests

any updates?

**Comment by asahalyft** [comment]
@terrytangyuan @merlintang I am facing the same set of problem as I try to incorporate the xgboost operator in our setup. Using the operator causes it to expect the CRD to be `v1` whereas the existing yaml is `v1alpha1`. But upgrading the CRD to `v1` causes thee `Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 433 [running]:`

Also, these instructions from README is also broken I believe. The folder structures have changed.

```
git clone https://github.com/kubeflow/manifests
cd manifests/xgboost-job/xgboost-operator
kubectl create namespace kubeflow
kustomize build base | kubectl apply -f -
```

should we downgrade to kubeflow manifest version `0.7` and stick to `v1alpha1`. If that is the case what should I checkout so the I have `v1alpha` compatible code for xgboost operator?

cc @Jeffwan

**Comment by merlintang** [comment]
@Jeffwan hi，jeff, can you update this as well? thanks so much

**Comment by asahalyft** [comment]
@merlintang @terrytangyuan Hello, I saw this fix https://github.com/kubeflow/xgboost-operator/pull/112 was merged. Is there any plan to publish the lates image to https://console.cloud.google.com/gcr/images/kubeflow-images-public/GLOBAL/xgboost-operator?gcrImageListsize=30 so that we can try out?

**Comment by terrytangyuan** [comment]
I don't think we are still updating images there as the test infra has changed quite a bit.  For now you should be able to build the image yourself using this Dockerfile: https://github.com/kubeflow/xgboost-operator/blob/master/Dockerfile

**Comment by Jeffwan** [comment]
@merlintang I missed the message. this has been fixed in #116 and #118  and I build a new image which works with current manifest. 

For people who has the issue, please use latest image and manifest

**Comment by Jeffwan** [comment]
I will close this since it's patched. Feel free to reopen if you still see problems
